runtime/lua/vim/lsp/linked_editing_range.lua

```diff
@@ -0,0 +1,177 @@
+local util = require('vim.lsp.util')
+local log = require('vim.lsp.log')
+local ms = require('vim.lsp.protocol').Methods
+local Range = require('vim.treesitter._range')
+local api = vim.api
+local M = {}
+
+---@class (private) vim.lsp.linked_editing_range.state Global state for linked editing ranges
+---@field enabled boolean Whether linked editing ranges are enabled
+---@field word_pattern? string
+---@field range_index? integer
```

Comment:
For maintainability purposes it would be nice to include descriptions for these.